extensions/topic_view_extension.rb

```diff
@@ -20,5 +20,73 @@ def self.included(base)
         base.const_set :ACTIVITY_FILTER, "activity"
       end
     end
+
+    # Moneky patch core's method. In an ideal world, we wouldn't have to do this here but `TopicView` in core does not
+    # yet properly support ordering posts in any other order except by `Post#sort_order` and several methods in
+    # core's `TopicView` is strongly tied to the assumption that posts are always ordered by `Post#sort_order`. Fixing
+    # core is hard and risky so we will just carry this monkey patch instead. There is also a small performance tradeoff
+    # here in the following implementation. In PostgreSQL, we basically have to scan for every single posts in order
+    # to figure out what the "row_number" for each post. From there, we can then properly fetch the window of posts
+    # near a given post number.
+    #
+    # Another alternative to this solution here is to rely on `Post#sort_order` and update the column of all posts
```

Comment:
Minor note, I don't think it's useful to describe an alternative approach in this code comment. I would just remove it from here (but keep it in the PR description).